osu.Game/OsuGameBase.cs

```diff
@@ -626,15 +625,40 @@ private void onRulesetChanged(ValueChangedEvent<RulesetInfo> r)
                 return;
             }
 
-            var previouslySelectedMods = SelectedMods.Value.ToArray();
+            var previouslySelectedCommonMods = new List<Mod>(SelectedMods.Value.Count);
+            var convertedCommonMods = new List<Mod>(SelectedMods.Value.Count);
+
+            foreach (var mod in SelectedMods.Value)
+            {
+                var convertedMod = instance.CreateModFromAcronym(mod.Acronym);
+
+                if (convertedMod == null)
+                    continue;
+
+                previouslySelectedCommonMods.Add(mod);
+
+                convertedMod.CopyCommonSettings(mod);
+                convertedCommonMods.Add(convertedMod);
+            }
+
+            if (!ModUtils.CheckValidForGameplay(convertedCommonMods, out var invalid))
+            {
+                invalid.ForEach(mod =>
+                {
+                    int index = convertedCommonMods.IndexOf(mod);
+                    convertedCommonMods.RemoveAt(index);
+                    previouslySelectedCommonMods.RemoveAt(index);
+                });
+            }
 
             if (!SelectedMods.Disabled)
-                SelectedMods.Value = Array.Empty<Mod>();
+                // Select common mods to play the deselect samples for other mods
+                SelectedMods.Value = previouslySelectedCommonMods;
```

Comment:
What on earth is this?

Reply:
It selects only the common mods so that the uncommon mods play their deselect sample and there isn't an audio discrepancy.

Reply:
I mean, that's what the comment says basically, but I still don't understand it at all.